pkg/defkinds/webserver/builder_test.go

```diff
@@ -233,9 +233,73 @@ func initBuildLLBForAlpineBasedBaseImgeTC(t *testing.T, mockCtrl *gomock.Control
 	}
 }
 
+func initBuildLLBWithCacheMountsTC(t *testing.T, mockCtrl *gomock.Controller) buildTC {
+	genericDef := loadGenericDef(t, "testdata/build/zbuild.yml")
+	genericDef.RawLocks = loadDefLocks(t, "testdata/build/zbuild.lock")
+
+	solver := mocks.NewMockStateSolver(mockCtrl)
+	kindHandler := webserver.WebserverHandler{}
+	kindHandler.WithSolver(solver)
+
+	return buildTC{
+		handler: &kindHandler,
+		client:  llbtest.NewMockClient(mockCtrl),
+		buildOpts: builddef.BuildOpts{
+			Def:               &genericDef,
+			Stage:             "",
+			SessionID:         "<SESSION-ID>",
+			LocalUniqueID:     "x1htr02606a9rk8b0daewh9es",
+			EnableCacheMounts: true,
```

Comment:
where are the cache mounts stored ? (in a cache image ?)
where is it represented in zbuild ? (I'm a bit lost in all these json files ^^)

Reply:
Well, Buildkit manages these cache mounts itself and it's not really documented so I still have some doubts about the whole thing :stuck_out_tongue:

> where are the cache mounts stored ? (in a cache image ?)

When you run `docker buildx build` (or `buildctl build`), you can pass `--cache-from` and `--cache-to` flags to tell Buildkit what it should use to retrieve and store the cache mounts. It can be either a local directory or an image (with a dedicated media type that don't match the one used for usual images, so only Docker Hub can store them for now iirc).

> where is it represented in zbuild ? (I'm a bit lost in all these json files ^^)

There're no representation of these cache mounts in zbuild, there's only a flag to toggle them. When that flag is true, [`llbutils.CacheMountOpt`](https://github.com/NiR-/zbuild/blob/0283765916c247d31413c2610b1f5c7686758f2e/pkg/llbutils/utils.go#L246) is called by specialized builders (or `llbutils.InstallSystemPackages` for instance) to add a `RunOpt` to `llb.Run(...)`. That `RunOpt` actually declares the mount.

In `CacheMountOpt`, the source is defined as `llb.Scratch()` (= a scratch image) with a `/cache` directory created with the given `owner`. And the mount is given an ID (the first argument of `llb.AsPersistentCacheDir()`).

From what I understand, when Buildkit detects a matching cache from `--cache-from` arguments, it discards the "from scratch" source and replaces it with the matching cache.